src/spn/algorithms/MEU.py

```diff
@@ -0,0 +1,162 @@
+"""
+Created on March 28, 2019
+@author: Hari Teja Tatavarti
+
+"""
+
+from spn.algorithms.MPE import get_node_funtions
+from spn.algorithms.Inference import  likelihood, max_likelihood, log_likelihood, sum_likelihood, prod_likelihood
+from spn.algorithms.Validity import is_valid
+from spn.structure.Base import get_nodes_by_type, Max, Leaf, Sum, Product, get_topological_order_layers
+
+import numpy as np
+from collections import defaultdict
+import collections
+
+
+def merge_input_vals(l):
+    return np.concatenate(l)
+
+def meu_max(node, parent_result, data=None, lls_per_node=None, rand_gen=None):
```

Comment:
Why does there have to be a `rand_gen` over here, it does not appear to be used in this function?

Reply:
Even here, I have taken the template from mpe_sum and added instructions for returning `decision_values` for that decision node. We may have to review mpe_sum and mpe_prod from MPE.py as well.